common/common.go

```diff
@@ -54,7 +54,15 @@ func initialiseHTTPClient() {
 // NewHTTPClientWithTimeout initialises a new HTTP client with the specified
 // timeout duration
 func NewHTTPClientWithTimeout(t time.Duration) *http.Client {
-	h := &http.Client{Timeout: t}
+	tr := &http.Transport{
+		// Added IdleConnTimeout to reduce the time of idle connections which
+		// could potentially slow macOS reconnection when there is a sudden
+		// network disconnection/issue
+		IdleConnTimeout: time.Second * 15,
```

Comment:
Very cool find, only request would be to use the t value so that the configurable value can be used for both idle connections and request timeouts